drivers/rtc/rtc_ll_stm32.c

```diff
@@ -67,7 +75,54 @@ LOG_MODULE_REGISTER(rtc_stm32, CONFIG_RTC_LOG_LEVEL);
 /* Timeout in microseconds used to wait for flags */
 #define RTC_TIMEOUT 1000000
 
+#ifdef CONFIG_RTC_ALARM
+#define RTC_STM32_ALRM_A	0U
+#define RTC_STM32_ALRM_B	1U
+
+#ifdef CONFIG_SOC_SERIES_STM32L4X
+#define RTC_EXTI_LINE	LL_EXTI_LINE_18
+#elif defined(CONFIG_SOC_SERIES_STM32C0X) \
+	|| defined(CONFIG_SOC_SERIES_STM32G0X)
+#define RTC_EXTI_LINE	LL_EXTI_LINE_19
+#elif defined(CONFIG_SOC_SERIES_STM32F4X) \
+	|| defined(CONFIG_SOC_SERIES_STM32F0X) \
+	|| defined(CONFIG_SOC_SERIES_STM32F1X) \
+	|| defined(CONFIG_SOC_SERIES_STM32F2X) \
+	|| defined(CONFIG_SOC_SERIES_STM32F3X) \
+	|| defined(CONFIG_SOC_SERIES_STM32F7X) \
+	|| defined(CONFIG_SOC_SERIES_STM32WBX) \
+	|| defined(CONFIG_SOC_SERIES_STM32G4X) \
+	|| defined(CONFIG_SOC_SERIES_STM32L0X) \
+	|| defined(CONFIG_SOC_SERIES_STM32L1X) \
+	|| defined(CONFIG_SOC_SERIES_STM32L5X) \
+	|| defined(CONFIG_SOC_SERIES_STM32H7X) \
+	|| defined(CONFIG_SOC_SERIES_STM32H5X) \
+	|| defined(CONFIG_SOC_SERIES_STM32WLX)
+#define RTC_EXTI_LINE	LL_EXTI_LINE_17
+#endif /* CONFIG_SOC_SERIES_STM32L4X */
+#endif /* CONFIG_RTC_ALARM */
+
+#if DT_INST_CLOCKS_CELL_BY_IDX(0, 1, bus) == STM32_SRC_LSI
+/* LSI */
+#define RTCCLK_FREQ STM32_LSI_FREQ
+#else
+/* LSE */
+#define RTCCLK_FREQ STM32_LSE_FREQ
+#endif /* DT_INST_CLOCKS_CELL_BY_IDX(0, 1, bus) == STM32_SRC_LSI */
+
+#if !defined(CONFIG_SOC_SERIES_STM32F1X)
+#define RTC_ASYNCPRE BIT_MASK(7)
+#else /* not CONFIG_SOC_SERIES_STM32F1X */
+#define RTC_ASYNCPRE (RTCCLK_FREQ - 1)
+#endif /* not CONFIG_SOC_SERIES_STM32F1X */
+
+/* Adjust the second sync prescaler to get 1Hz on ck_spre */
+#define RTC_SYNCPRE ((RTCCLK_FREQ / (1 + RTC_ASYNCPRE)) - 1)
+
 struct rtc_stm32_config {
+#ifdef CONFIG_RTC_ALARM
+	uint32_t alarms_count;
```

Comment:
A bit overkill with a u32 :) maybe uint8 or uint16 (to match with the size of id in `rtc_alarm_set_time`)

Reply:
Replaced by `RTC_STM32_ALARMS_COUNT` in `rtc_ll_stm32.h`